distributing_producer.go

```diff
@@ -17,7 +17,7 @@ import (
 // partition, returning the post-commit offset and any error encountered. The
 // offset of each message is also updated accordingly.
 type DistributingProducer interface {
-	Distribute(topic string, messages ...*proto.Message) (offset int64, err error)
+	Distribute(topic string, messages ...*proto.Message) (offset int64, partition int32, err error)
```

Comment:
Flip the order -- the partition is the "more important" bit so it should go first.